### PR TITLE
docs: data binding

### PR DESCRIPTION
**Pull request by Ombuweb**

### CODE REVIEW

**Comment by rigor789** [comment]
@cla-bot check

**Comment by CatchABus** [comment]
@Ombuweb I think we should stick with the term of view model instead of data model.
```
The term ViewModel originates from the MVVM design pattern. In the MVVM design pattern, it is the viewmodel that contains all the logic to handle the request/events generated by the view. We can say that a viewmodel in MVVM pattern is like a controller in MVC pattern.
```

File: content/guide/data-binding.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+It gets activated through three steps:
+
+1. Create a ViewModel(let's call it DataModel) class extending the `Observable` class
+2. Make DataModel available to the UI by setting `page.bindingContext` = `new DataModel()`
+3. Using the mustach syntax ({{ }} ), bind the UI components properties to the members of the `DataModel` instance.
+
+When you look at a new project, you see an example of those steps applied.
+
+There are several ways of data flow (data bindings):
```
Comment (SeanKelly369) [inline content/guide/data-binding.md]:
```suggestion
There are various ways to manage data flow, often referred to as data bindings:
```

File: content/guide/data-binding.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+- **One-way (to UI) data binding** - This is the most popular way of binding. A good example is bindingi a `string` property in the Model to a `Label` component.
```
Comment (SeanKelly369) [inline content/guide/data-binding.md]:
```suggestion
- **One-way (to UI) data binding** - This is the most common form of data binding. An example would be binding a string property in the Model to a Label component.
```

File: content/guide/data-binding.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+```
+```ts
+export class HelloWorldModel extends Observable {
+        name = "John Doe"
+
+        ...
+
+}
+```
+- **One-way (to Model) data binding** - Binding which updates Model due to some action on UI. The best example for this is a [tap](/guide/ui-components/gestures-in-nativescript#tap-gesture-in-nativescript) event.
```
Comment (SeanKelly369) [inline content/guide/data-binding.md]: What is the url to the tap linke?

Comment (SeanKelly369) [inline content/guide/data-binding.md]:
```suggestion
- **One-way (to Model) data binding** - Binding which updates the Model by triggering some type of action within the UI. The best example for this is a [tap](/guide/ui-components/gestures-in-nativescript#tap-gesture-in-nativescript) event.
```

File: content/guide/data-binding.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        ...
+
+}
+```
+
+## Accessing parent bindingContext
+
+A parent  and a child UI components can have different binding contexts and the child component might need to bind its property to a source property in its parent's **bindingContext**.
+
+<!-- TODO: fix links -->
```
Comment (SeanKelly369) [inline content/guide/data-binding.md:65]: Two TODOs here.  Adding this comment as a reminder.

File: content/guide/data-binding.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+::: tip Note
+If the value of the `items` property of the `ListView` is an array of plain elements(numbers,string, dates) as in the preceeding example, you use the `$value` variable to access the current item of the array.
+
+If it is an array of objects,you use the current object property name as the variable name.
+:::
+
+## Using binding expressions
+
+NativeScript supports the following polymer binding expressions :
```
Comment (SeanKelly369) [inline content/guide/data-binding.md]:
```suggestion
NativeScript supports the following polymorphic binding expressions:
```

File: content/guide/data-binding.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+- double quotes(`"`) => `&quot;`
+- single quote(`'`) => `&apos;`
+- less than(`<`) => `&lt;`
+- greater than(`>`) => `&gt;`
+- ampersand(`&`) => `&amp;`
+
+## Using data converters
+
+Often data within ViewModel is stored in a way that is optimized for best performance of tasks like search, replace and so on. Unfortunately, the way computers store data differs a lot with a human readable format. Probably the best example is the `Date` object. In JavaScript, Date actually is a very big number that represents milliseconds from `01.01.1970` which does not speak much to any human. Here comes the use of data converters which basically are functions that formats the data (from Model) in a human readable format (display in UI).
```
Comment (SeanKelly369) [inline content/guide/data-binding.md]:
```suggestion
Often data within the Model is stored in a way that is optimized for best performance regarding tasks such as search, replace etc. Unfortunately, the way computers store data differs significantly from human-readable formats. One fitting example is the Date object. In JavaScript, Date actually is a very big number that represents milliseconds from `01.01.1970` which does not speak much to any human. This is where data converters come into play. Data converters are essentially functions that format the data from the Model into a human-readable format for display in the UI.
```

File: content/guide/data-binding.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+  ... 
+
+formatDate() {
+    return {
+      toView(value: number, format: string) {
+
+        const date = new Date(value)
+        const day = date.getDate().toString().padStart(2, "0")
+        const month = (date.getMonth() + 1).toString().padStart(2, "0")
```
Comment (SeanKelly369) [inline content/guide/data-binding.md]:
```suggestion
        const month = (date.getMonth() + 1).toString().padStart(2, "0") // months are zero based in JS.
```

File: content/guide/data-binding.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+  <Label text="{{ date | dateConverter('dd-mm-yyyy') }}" textWrap="true" />
+  <Label text="{{ name | toUpperCaseConverter }}" textWrap="true" />
+  <Label text="{{ title | toTitle }}" textWrap="true" />
+
+  </StackLayout>
+
+ ```
+
+```ts
+export class HelloWorldModel extends Observable {
```
Comment (SeanKelly369) [inline content/guide/data-binding.md:225]: Would it be better practices to add a ; where applicable, even if in JS it is optional?

File: content/guide/data-binding.md (hunk 3 above)
Comment (NathanWalker) [inline content/guide/data-binding.md]:
```suggestion
- **One-way (to Model) data binding** - Binding which updates the model in relation to some action in the UI. The best example for this is a [tap](/guide/ui-components/gestures-in-nativescript#tap-gesture-in-nativescript) event.
```

File: content/guide/data-binding.md [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+        onTap(args: EventData){
+
+        }
+
+        ...
+
+}
+```
+
+- **Two-way data binding** - The data flows from the Model to the UI and vice versa. A typical example is a `TextField` that reads its value from Model, and also changes the Model based on user input.
```
Comment (NathanWalker) [inline content/guide/data-binding.md]:
```suggestion
- **Two-way data binding** - The data flows from the Model to the UI and vice versa. A typical example is a `TextField` that reads its value from a Model, and also changes the Model based on user input.
```

**Pull request merged**